### ISSUE TITLE
Automapper equivalent to "As"

### ISSUE THREAD

**Issue by BlowaXD** [MapsterMapper/Mapster#203]
Hi,

Thanks for your work on this amazing lib :)

I just moved all my project from AutoMapper to Mapster, now I've got small issues.

I would like to know if there is some kind of alternative for `.As<TObject>()` in AutoMapper

https://docs.automapper.org/en/stable/Mapping-inheritance.html?highlight=.As%3C#as

I'm using it this way on automapper
![image](https://user-images.githubusercontent.com/16936368/62014088-ef574800-b19c-11e9-9804-04780c55472b.png)

Here is an object that is
![image](https://user-images.githubusercontent.com/16936368/62014098-072ecc00-b19d-11e9-8101-2ae51059d5a7.png)

I'm mapping from my Database Entities to my GameObjects but it's used as DTOs.
This is the mapping Configuration I was trying to get working as I expected
![image](https://user-images.githubusercontent.com/16936368/62014112-2af21200-b19d-11e9-84d8-8298af4b8e52.png)
Here goes the error
![image](https://user-images.githubusercontent.com/16936368/62014120-3ba28800-b19d-11e9-9fe4-6f9067e5b462.png)

Here is the mapping code

Basically, I would like to override the destination type to the GameObject (which always inherits from the DTO)
![image](https://user-images.githubusercontent.com/16936368/62014146-8c19e580-b19d-11e9-941a-7bbbefb9fc49.png)

Thanks by advance !

**Comment by chaowlert** [comment]
What you did is correct, `Mapster` doesn't provide method equivalent to `As`. 

If you use `ConstructUsing`, You can just create target instance.

```
TypeAdapterConfig<Recipe, RecipeDTO>.NewConfig()
    .ConstructUsing(s => new Game.Recipe());
```

Hope this help.

**Comment by BlowaXD** [comment]
Going to take a look tomorrow if I have time :)

**Comment by chaowlert** [comment]
Please also try `Include`.

```
TypeAdapterConfig<Recipe, RecipeDTO>.NewConfig().Include<Recipe, Game.Recipe>();
```

**Comment by BlowaXD** [comment]
only using ConstructUsing didn't work, going to check with include :+1:

**Comment by BlowaXD** [comment]
Using .Include<Recipe, Game.Recipe>(); does not work too

**Comment by BlowaXD** [comment]
Strangely, this work in the following unit tests
```cs
[Fact]
        public void Check_Map_To_ChildClass()
        {
            MappingInitializer.InitializeMapping();
            short recipeId = 1;
            MapsterMapper<Recipe, RecipeDTO> mapper = new MapsterMapper<Recipe, RecipeDTO>();

            Check.That(mapper.MapToDto(new Recipe { RecipeId = recipeId }, out RecipeDTO dto)).IsTrue();

            var recipe = (WingsEmu.Game.Recipe)dto;

            Check.That(dto).IsInstanceOf<WingsEmu.Game.Recipe>();
            Check.That(dto.RecipeId).IsEqualTo(recipeId);
        }

        [Fact]
        public void Check_Map_To_ChildClasses()
        {
            MappingInitializer.InitializeMapping();
            short recipeId = 1;
            MapsterMapper<Recipe, RecipeDTO> mapper = new MapsterMapper<Recipe, RecipeDTO>();

            List<Recipe> list = new List<Recipe>
            {
                new Recipe { RecipeId = recipeId }, new Recipe { RecipeId = recipeId }
            };
            Check.That(mapper.MapToDtos(list, out IEnumerable<RecipeDTO> dtos)).IsTrue();


            foreach (RecipeDTO dto in dtos)
            {
                var obj = (WingsEmu.Game.Recipe)dto;
                Check.That(dto).IsInstanceOf<WingsEmu.Game.Recipe>();
                Check.That(dto.RecipeId).IsEqualTo(recipeId);
            }
        }
```
xUnit, NFluent

**Comment by BlowaXD** [comment]
Fixed, I got some forgotten configuration on IoC :(
